state/state.go

```diff
@@ -10,6 +10,9 @@ import (
 	"encoding/binary"
```

Comment:
I moved "Pub" out of state to a separate pubregistry package/struct because the growing number of publishers in the registry started to pollute the already severely polluted state package. (And vice versa)
Placing it in the pubsub package will cause import cycle errors, plus a surplus is that in IntelliJ/golang the drop-down list for state.Pub. is cvery clean.